kolibri/plugins/management/assets/src/state/facilityConfigPageActions.js

```diff
@@ -0,0 +1,88 @@
+/* eslint-disable prefer-arrow-callback */
+const omit = require('lodash/fp/omit');
+const {
+  FacilityResource,
+  FacilityDatasetResource,
+} = require('kolibri').resources;
+const ConditionalPromise = require('kolibri.lib.conditionalPromise');
+const { samePageCheckGenerator } = require('kolibri.coreVue.vuex.actions');
+const preparePage = require('./preparePage');
+const { PageNames, defaultFacilityConfig, notificationTypes } = require('../constants');
+
+// When app is installed, the Facility is assigned id of `1`. Hardcoded here for now.
+const FACILITY_ID = 1;
+
+function resolveOnlyIfOnSamePage(promises, store) {
+  const ident = x => x;
+  return ConditionalPromise.all(promises)
+  .only(samePageCheckGenerator(store), ident, ident)._promise;
+}
+
+const sanitizeDataset = omit(['id']);
+
+function showNotification(store, notificationType) {
+  store.dispatch('CONFIG_PAGE_NOTIFY', notificationType);
+}
+
+function showFacilityConfigPage(store) {
+  preparePage(store.dispatch, {
+    name: PageNames.FACILITY_CONFIG_PAGE,
+    title: 'Configure Facility',
+  });
+  const resourceRequests = [
+    FacilityResource.getModel(FACILITY_ID).fetch(),
+    FacilityDatasetResource.getCollection({ facility_id: FACILITY_ID }).fetch(),
+  ];
+
+  return resolveOnlyIfOnSamePage(resourceRequests, store)
+  .then(function onSuccess([facility, facilityDataset]) {
+    const dataset = facilityDataset[0]; // assumes for now is only one Facility being managed
+    store.dispatch('SET_PAGE_STATE', {
+      // comes over wire as number, but gets changed to string
+      facilityDatasetId: Number(dataset.id),
+      facilityName: facility.name,
+      // this part of state is mutated as user interacts with form
+      settings: sanitizeDataset(dataset),
```

Comment:
In other places, we've used "API data to client state" mapping functions to explicitly map/pluck the values returned from the API to vuex state keys.

Even though simply dropping the contents of the API response into vuex is concise, I think it's a little messier long-term because:

 * there's no explicit place in js where the vuex state attributes get defined - instead it ends up implicitly defined in the vue files
 * server-side naming conventions (e.g. the underscores in `learner_can_edit_name`) start spilling into client-side code